services/config_data_sources/json_project_data_source.rb

```diff
@@ -158,16 +158,22 @@ def save_git_repo_configs!(git_repo_configs: nil)
     end
 
     def create_project!(name: nil, repo_config: nil, enabled: nil, lane: nil, artifact_provider: nil)
-      projects = self.projects
+      projects = self.projects.clone
       new_project = Project.new(repo_config: repo_config,
                                 enabled: enabled,
                                 project_name: name,
                                 lane: lane,
                                 artifact_provider: artifact_provider)
       if !self.project_exist?(new_project.project_name)
-        projects.push(new_project)
+        projects << new_project
         self.projects = projects
         logger.debug("Added project #{new_project.project_name} to projets.json in #{self.json_folder_path}")
+        logger.debug("Trigger clone project #{new_project.project_name}")
+        _ = GitRepo.new(
```

Comment:
Sorry I didn't notice this until now, but I don't think this is the right location to be creating a GitRepo, or cloning. This data source is for managing the `Project` object.

Reply:
Where should be the best place to clone the repo associated to the project? I mean, is necessary for a project to work to clone it in create time or it can be delayed?

Reply:
Anyway, I really think that is not even necessary to clone the git repo of the project, as It'll be done when a build is triggered...

Reply:
I'm afraid that on project settings, as the repo may not being cloned yet, we won't have access to the repo's lanes